### ISSUE TITLE
Protect Japonicus web UI with username+password?

### ISSUE THREAD

**Issue by rcoenen** [Gab0/japonicus#185]
Now that I have everything up & running on an EC2 instance:

I have the nice web interface running for Japonicus. Can this protected with a simple username / password (basic auth)

**Comment by Gab0** [comment]
Flask should have some option for it, but I don't know...
Try to do a research and update the repo :3 If it's built in on Flask it should be simple.

**Comment by rcoenen** [comment]
Yeah I have been messing around with that for a few hours. I'm earning Flask, Dash, Python, etc here in real-time.
I guess I should try to hook in the basic auth in `/japonicus/promoterz/webServer/core.py` somewhere. Any pointers, code snippets?

**Comment by rcoenen** [comment]
Ok update: I have it working now. Turns out basic auth had to be added to the Dash app, not the Flash server (or in either case, this is how I got it to work)
1) yes the file to modify is `/japonicus/promoterz/webServer/core.py`
2) helpful example found here https://dash.plot.ly/authentication -- see Basic Auth Example
